Open-Rival/include/Unit.h

```diff
@@ -3,8 +3,6 @@
 
 #include "Spritesheet.h"
 #include "UnitAnimation.h"
-#include "UnitAnimationLookup.h"
-#include "UnitTypes.h"
```

Comment:
Unit.h includes UnitAnimation.h, which includes UnitAnimationLookup.h and UnitTypes.h. Doing this way will let us know (with a compilation error) when something changes about our structure.

Reply:
My understanding is that it is best practice to always include headers needed by the current file, rather than relying on transitive includes.

Some more on this subject here:
https://softwareengineering.stackexchange.com/questions/262019/is-it-good-practice-to-rely-on-headers-being-included-transitively

Reply:
From provided link:

> This makes it clear to developers who read the source exactly what the source file in question requires. Here, someone looking at the first few lines in the file can see that you are dealing with Texture objects in this file.

fair enough, but

> This avoids issues where refactored headers cause compilation issues when they no longer require particular headers themselves. For instance, suppose you realize that RenderObject.hpp doesn't actually need Texture.hpp itself.

This is exactly what we want. We do not want bugs to make their way into production code silently. We want compilation crashes. Although I have strong opinions about these, this is your code, and I've reverted :)

Reply:
I'm always up for a discussion - I don't want to just dictate everything, especially as I'm still a C++ novice 😄 

It's just that I have in my head certain "rules" that I've learnt, so when I see one of these rules being broken I want to understand why.

I agree that omitting these headers would cause compilation headers if they were removed from the `UnitAnimation.h`, but what are the silent bugs you're referring to? If `Unit.h` includes these headers directly, then what would break?